execution.bs

```diff
@@ -4108,28 +4111,219 @@ namespace std::execution {
   using <i>execute</i>::execute_t;
   inline constexpr execute_t execute{};
 
-  // [exec.as_awaitable]
+  // [exec.as.awaitable]
   namespace <i>coro-utils</i> { // exposition only
     struct as_awaitable_t;
   }
   using <i>coro-utils</i>::as_awaitable_t;
   inline constexpr as_awaitable_t as_awaitable;
 
-  // [exec.with_awaitable_senders]
+  // [exec.with.awaitable.senders]
   template &lt;<i>class-type</i> Promise>
     struct with_awaitable_senders;
 }
 </pre>
 
-## Helper concepts <b>[exec.helpers]</b> ## {#spec-execution.helpers}
+## Helper entities <b>[exec.helpers]</b> ## {#spec-execution.helpers}
+
+1. This clause makes use of the following exposition-only entities.
 
     <pre highlight=c++>
-    template&lt;class T>
+    template &lt;class Fn, class... Args>
+        requires <i>callable</i>&lt;Fn, Args...>
+      constexpr decltype(auto) <i>mandate-nothrow-call</i>(Fn&& fn, Args&&... args) noexcept {
+        return std::forward<Fn>(fn)(std::forward<Args>(args)...);
+      }
+    </pre>
+
+    * <i>Mandates:</i> <code><i>nothrow-callable</i>&lt;Fn, Args...></code> is `true`.
+
+    <pre highlight=c++>
+    template &lt;class T>
       concept <i>movable-value</i> = <i>// exposition only</i>
         move_constructible&lt;decay_t&lt;T>> &&
         constructible_from&lt;decay_t&lt;T>, T>;
     </pre>
 
+    For function types `F1` and `F2` denoting `R1(Args1...)` and `R2(Args2...)`
+    respectively, <code><i>MATCHING-SIG</i>(F1, F2)</code> is `true` if and only if 
+    `same_as<R1(Args&&...), R2(Args2&&...)>` is `true`.
+
+## Asynchronous functions <b>[async.func]</b> ## {#spec-execution-async.func}
+
+Subsections [exec.queries] through [exec.snd] specify a set of interrelated concepts
+for defining asynchronous algorithms. This subsection defines some terms and then
+specifies the [=async function protocol=] that asynchronous algorithms must
+adhere to.
+
+### Terms and definitions <b>[async.defn]</b> ### {#spec-execution-async.defn}
```

Comment:
ISO brain-damage: don't call this terms and definitions. Definitions must be macro-replaceable with the terms.